src/cli/serve/integration_tests/reload_logging_config.test.js

```diff
@@ -80,66 +78,68 @@ describe('Server logging configuration', function () {
-        expect(sawNonjson).toEqual(true);
-        done();
-      });
-
-      child.stdout
-        .pipe(es.split())
-        .pipe(es.mapSync(line => {
-          if (!line) {
-            // skip empty lines
-            return;
-          }
-
-          if (isJson) {
-            const data = JSON.parse(line);
-            sawJson = true;
-
-            if (data.tags.includes('listening')) {
-              switchToPlainTextLog();
+      const [exitCode] = await Promise.all([
+        Promise.race([
+          new Promise(r => child.once('exit', r))
+            .then(code => code === null ? 0 : code),
+
+          new Promise(r => child.once('error', r))
+            .then(err => {
+              throw new Error(`error in child process while attempting to reload config. ${err.stack || err.message || err}`);
+            })
+        ]),
+
+        createPromiseFromStreams([
+          child.stdout,
+          createSplitStream('\n'),
+          createMapStream(async (line) => {
+            if (!line) {
+              // skip empty lines
+              return;
             }
-          } else if (line.startsWith('{')) {
-            // We have told Kibana to stop logging json, but it hasn't completed
-            // the switch yet, so we ignore before switching over.
-          } else {
-            // Kibana has successfully stopped logging json, so kill the server.
-
-            sawNonjson = true;
 
-            child.kill();
-            child = undefined;
-          }
-        }));
-
-      function switchToPlainTextLog() {
-        isJson = false;
-        setLoggingJson(false);
-
-        // Reload logging config. We give it a little bit of time to just make
-        // sure the process sighup handler is registered.
-        setTimeout(() => {
-          child.kill('SIGHUP');
-        }, 100);
-      }
-    }, minute);
+            if (isJson) {
+              const data = JSON.parse(line);
+              sawJson = true;
+
+              if (data.tags.includes('listening')) {
+                isJson = false;
+                setLoggingJson(false);
+
+                // Reload logging config. We give it a little bit of time to just make
+                // sure the process sighup handler is registered.
+                await new Promise(r => setTimeout(r, 100));
+                child.kill('SIGHUP');
+              }
+            } else if (line.startsWith('{')) {
+              // We have told Kibana to stop logging json, but it hasn't completed
+              // the switch yet, so we ignore before switching over.
+            } else {
+              // Kibana has successfully stopped logging json, so kill the server.
+
+              sawNonjson = true;
+
+              child.kill();
+              child = undefined;
+            }
+          })
+        ])
+      ]);
+
+      expect(exitCode).toEqual(0);
+      expect(sawJson).toEqual(true);
+      expect(sawNonjson).toEqual(true);
+    }, 60000);
```

Comment:
__optional nit:__ there should a `minute`, but I don't want you to wait another CI run, so I'm fine with backporting this as is.

Reply:
Coolio, going to merge this!